src/components/Navigation.tsx

```diff
@@ -9,30 +9,32 @@ import ServerVersion from "components/ServerVersion";
 import useEventListener from "@use-it/event-listener";
 import { isWidthBelow } from "util/helpers";
 import { useProject } from "context/project";
+import { useMenuCollapsed } from "context/menuCollapsed";
 
 const isSmallScreen = () => isWidthBelow(620);
 
 const Navigation: FC = () => {
-  const [menuCollapsed, setMenuCollapsed] = useState(isSmallScreen());
-
+  const { menuCollapsed, setMenuCollapsed } = useMenuCollapsed();
   const { project, isLoading } = useProject();
 
   const projectName = project && !isLoading ? project.name : "default";
 
   const { isAuthenticated } = useAuth();
   const softToggleMenu = () => {
     if (isSmallScreen()) {
-      setMenuCollapsed((prev) => !prev);
+      setMenuCollapsed(!menuCollapsed);
     }
   };
 
   const hardToggleMenu = (e: MouseEvent<HTMLElement>) => {
-    setMenuCollapsed((prev) => !prev);
+    setMenuCollapsed(!menuCollapsed);
```

Comment:
The previous style with the callback is safer, because `menuCollapsed` comes from the wrapping component and is not guaranteed to be up to date, while the `prev` argument the callback provides comes with that guarantee.

Reply:
Right, I was experimenting one thing and then I forgot to revert this change. It will be back as it was before.